README.rst

```diff
@@ -1198,6 +1245,181 @@ Once you have performed the 3 steps above, you can test it as follows:
 5. Shortly after you change the name in OpenWISP, you should see some output in the
    SSH console indicating another SSH access and the configuration update being performed.
 
+Sending Commands to Devices
+---------------------------
+
+By default, there are three options in the **Send Command** dropdown:
+
+1. Reboot
+2. Change Password
+3. Custom Command
+
+While the first two options are self-explanatory, the **custom command** option
+allows you to execute any command on the device as shown in the example below.
+
+.. image:: https://raw.githubusercontent.com/openwisp/openwisp-controller/docs/docs/commands_demo.gif
+   :target: https://github.com/openwisp/openwisp-controller/tree/docs/docs/commands_demo.gif
+   :alt: Executing commands on device example
+
+**Note**: A device needs to have at least one **Access Credential**, in order for
+this feature to work. **Send Command** button will be hidden until the device
+has at least one **Access Credential**.
+
+If you need to allow your users to quickly send specific commands that are used often in your
+network regardless of your users' knowledge of Linux shell commands, you can add new commands
+by following instructions in `"How to add commands" <#how-to-add-commands>`_ section.
+
+If you are an advanced user and want to register commands programatically, then refer to
+`"Register / Unregistering commands" <#registering--unregistering-commands>`_ section.
+
+How to add commands
+~~~~~~~~~~~~~~~~~~~
+
+This example introduces a simple command that could ``ping`` an input
+``destination_address`` through an interface, ``interface_name``.
+
+.. code-block:: python
+
+    OPENWISP_CONTROLLER_USER_COMMANDS = [
+        (
+            'ping',
+            {
+                'label': 'Ping',
+                'schema': {
+                    'title': 'Ping',
+                    'type': 'object',
+                    'required': ['destination_address'],
+                    'properties': {
+                        'destination_address': {
+                            'type': 'string',
+                            'title': 'Destination Address',
+                        },
+                        'interface_name': {
+                            'type': 'string',
+                            'title': 'Interface Name',
+                        },
+                    },
+                    'message': 'Destination Address cannot be empty',
+                    'additionalProperties': False,
+                },
+                'callable': 'path.to.command.callable',
```

Comment:
can you include also the callable in the example? Either in the same file and pass the function directly or in another example file, passing the right path to it, but the simpler the better, advanced users will know what to do, less advanced users may just copy/paste the example and be happy with it.